esmf_regrid/esmf_regridder.py

```diff
@@ -76,8 +84,9 @@ def __init__(self, src, tgt, method="conservative", precomputed_weights=None):
             Describes the target mesh/grid.
             Data output by this regridder will be a :class:`numpy.ndarray` whose
             shape is compatible with ``tgt``.
-        method : str
-            Either "conservative", "bilinear" or "nearest". Corresponds to the :mod:`esmpy` methods
+        method : :class:`Regridder.Method`
+            Either Regridder.Method.CONSERVATIVE, Regridder.Method.BILINEAR or Regridder.Method.NEAREST.
+            Corresponds to the :mod:`esmpy` methods
             :attr:`~esmpy.api.constants.RegridMethod.CONSERVE`,
             :attr:`~esmpy.api.constants.RegridMethod.BILINEAR` or
             :attr:`~esmpy.api.constants.RegridMethod.NEAREST_STOD` used to calculate weights.
```

Comment:
One of the appeals of using `Enum` is being able to avoid listing things within other docstrings. ``:class:`Regridder.Method` `` renders as a nice link to your new enumeration; we could put the bulk of the documentation within `Method` instead, just leaving basic info behind.

![image](https://github.com/SciTools-incubator/iris-esmf-regrid/assets/40734014/f052ac52-40eb-4ce3-8672-e4a6572de708)

Reply:
```suggestion
        method : :class:`Regridder.Method`
            The method to be used to calculate weights.
```